context/AuthContextProvider.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+import { createContext, useContext, useEffect, useMemo, useState } from 'react';
+import { account } from '@/api/config';
+import { useRouter } from 'next/navigation';
+
+type UserCredentials = {
+  email: string;
+  password: string;
+};
+
+type AuthContextType = {
+  isSignedIn: boolean;
+  setIsSignedIn: (isSignedIn: boolean) => void;
+  loginAccount: (user: UserCredentials) => Promise<void | Error>;
+  logoutAccount: () => Promise<void>;
+};
+
+export const AuthContext = createContext<AuthContextType | null>(null);
+
+export const AuthContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);
+  const router = useRouter();
+
+  // Check for a current session on component mount
```

Comment:
I'm ok with these comments for now until we add JSDocs.

Reply:
cool beans.